mmseg/apis/ote/apis/segmentation/inference_task.py

```diff
@@ -186,67 +188,58 @@ def hook(module, input, output):
 
-        prediction_results, _ = self._infer_segmentor(self._model, self._config, dataset, eval=False,
-                                                      output_logits=True, dump_features=True)
-
-        label_dictionary = {
-            i + 1: self._labels[i] for i in range(len(self._labels))
-        }
-
-        # Loop over dataset again to assign predictions. Convert from MMSegmentation format to OTE format
-        for dataset_item, (soft_prediction, feature_vector) in zip(dataset, prediction_results):
-            soft_prediction = np.transpose(soft_prediction, axes=(1, 2, 0))
-
-            hard_prediction = create_hard_prediction_from_soft_prediction(
-                soft_prediction=soft_prediction,
-                soft_threshold=self._hyperparams.postprocessing.soft_threshold,
-                blur_strength=self._hyperparams.postprocessing.blur_strength,
-            )
-
-            annotations = create_annotation_from_segmentation_map(
-                hard_prediction=hard_prediction,
-                soft_prediction=soft_prediction,
-                label_map=label_dictionary,
-            )
-
-            dataset_item.append_annotations(annotations=annotations)
-
-            if feature_vector is not None:
-                active_score = TensorEntity(name="representation_vector", numpy=feature_vector)
-                dataset_item.append_metadata_item(active_score, model=self._task_environment.model)
-
-            if dump_features:
-                for label_index, label in label_dictionary.items():
-                    if label_index == 0:
-                        continue
-
-                    if len(soft_prediction.shape) == 3:
-                        current_label_soft_prediction = soft_prediction[:, :, label_index]
-                    else:
-                        current_label_soft_prediction = soft_prediction
-
-                    min_soft_score = np.min(current_label_soft_prediction)
-                    max_soft_score = np.max(current_label_soft_prediction)
-                    factor = 255.0 / (max_soft_score - min_soft_score + 1e-12)
-                    result_media_numpy = (factor * (current_label_soft_prediction - min_soft_score)).astype(np.uint8)
-
-                    result_media = ResultMediaEntity(name=f'{label.name}',
-                                                     type='Soft Prediction',
-                                                     label=label,
-                                                     annotation_scene=dataset_item.annotation_scene,
-                                                     roi=dataset_item.roi,
-                                                     numpy=result_media_numpy)
-                    dataset_item.append_metadata_item(result_media, model=self._task_environment.model)
+        self._infer_segmentor(self._model, self._config, dataset,
+                              output_logits=True, dump_features=True, add_predictions_to_dataset=True,
```

Comment:
In which case predictions are not saved to the dataset or features are not dumped or logits are not needed?

Reply:
removed extra parameters